cider/type/plan/BinaryExpr.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "BinaryExpr.h"
+#include "exec/template/Execute.h"  // for is_unnest
+
+namespace Analyzer {
+using namespace cider::jitlib;
+
+JITExprValue& BinOper::codegen(JITFunction& func) {
```

Comment:
These files are related to codegen, and it is not appropriate to put them in the plan directory

Reply:
then should we move all Analyzer out of plan?

Reply:
Hmm, will be better to decouple plan definition and codegen logic? Saying ```ColumnVar``` with ```ColumnVarCodegen```. Thoughts?

Reply:
we may reconsider how to refactor this API after the nextgen is finished.